File: src/main/java/hoten/serving/JsonMessageBuilder.java
package hoten.serving;

import com.google.gson.Gson;
import hoten.serving.Protocols.Protocol;
import java.io.UnsupportedEncodingException;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Level;
import java.util.logging.Logger;

public class JsonMessageBuilder {
    private final Map _map = new HashMap();
    private Protocol _protocol;
    
    public JsonMessageBuilder set(String key, Object value) {
        _map.put(key, value);
        return this;
    }
    
    public JsonMessageBuilder protocol(Protocol protocol) { 
        _protocol = protocol;
        return this;
    }
    
    public Message build() {
        byte[] data = null;
        try {
            data = new Gson().toJson(_map, Map.class).getBytes("UTF-8");
        } catch (UnsupportedEncodingException ex) {
            Logger.getLogger(JsonMessageBuilder.class.getName()).log(Level.SEVERE, null, ex);
        }
        return Message.OutboundMessage(_protocol, data);
    }
}

File: src/main/csharp/Serving/Serving/JsonMessageBuilder.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace Serving
{
    public class JsonMessageBuilder
    {
        private Dictionary<String, Object> _map = new Dictionary<String, Object>();
        private Protocol _protocol;

        public JsonMessageBuilder Set(String key, Object value)
        {
            _map[key] = value;
            return this;
        }

        public JsonMessageBuilder Protocol(Protocol protocol)
        {
            _protocol = protocol;
            return this;
        }

        public Message Build()
        {
            var json = JsonConvert.SerializeObject(_map);
            var data = Encoding.UTF8.GetBytes(json);
            return Message.OutboundMessage(_protocol, data);
        }
    }
}
